internal/bminventory/inventory.go

```diff
@@ -4110,6 +4112,29 @@ func (b *bareMetalInventory) getHost(ctx context.Context, clusterId string, host
 	return &host, nil
 }
 
+func (b *bareMetalInventory) GetHostInternal(ctx context.Context, clusterId string, hostId string) (*common.Host, error) {
+	log := logutil.FromContext(ctx, b.log)
+	var host common.Host
+
+	if err := b.db.First(&host, "id = ? and cluster_id = ?", hostId, clusterId).Error; err != nil {
+		log.WithError(err).Errorf("failed to find host: %s", hostId)
+		return nil, common.NewApiError(http.StatusNotFound, errors.Errorf("Host %s not found", hostId))
+	}
+	return &host, nil
+}
+
+func (b *bareMetalInventory) UpdateHostApprovedInternal(ctx context.Context, clusterId string, hostId string, approved bool) error {
+	log := logutil.FromContext(ctx, b.log)
+	var host common.Host
+
+	log.Infof("Updating Approved to %t Host %s Cluster %s", approved, hostId, clusterId)
+	if err := b.db.First(&host, "id = ? and cluster_id = ?", hostId, clusterId).Error; err != nil {
+		log.WithError(err).Errorf("failed to find host: %s", hostId)
+		return common.NewApiError(http.StatusNotFound, errors.Errorf("Host %s not found", hostId))
+	}
+	return b.db.Model(&common.Host{}).Where(identity.AddUserFilter(ctx, "id = ? and cluster_id = ?"), hostId, clusterId).Update("approved", approved).Error
```

Comment:
don't we want to log the error if occurred?

Reply:
Done